### PR TITLE
chore: upgrade typescript and eslint

### PR DESCRIPTION
**Pull request by ekashida**
## Details

Upgrade typescript and eslint

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

File: packages/@lwc/compiler/src/rollup-plugins/module-resolver.ts [hunk 1]
```diff
@@ -45,14 +46,14 @@ function isFirstCharacterUppercased(importee: string) {
 
 function inferExtension(fileName: string, files: BundleFiles) {
     if (!path.extname(fileName)) {
-        const ext = VALID_EXTENSIONS.find(ext => files.hasOwnProperty(fileName + ext)) || '';
```
Comment (caridy) [inline packages/@lwc/compiler/src/rollup-plugins/module-resolver.ts]: on utilities like this, that are not subject to poisoning, we don't need to caching... but I'm fine either way.

File: packages/integration-karma/test/polyfills/aria-properties/index.spec.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 function testAriaProperty(property, attribute) {
     describe(property, () => {
         it(`should assign property ${property} to Element prototype`, () => {
-            expect(Element.prototype.hasOwnProperty(property)).toBe(true);
+            expect(Object.prototype.hasOwnProperty.call(Element.prototype, property)).toBe(true);
```
Comment (caridy) [inline packages/integration-karma/test/polyfills/aria-properties/index.spec.js]: why was this a problem?

File: packages/@lwc/compiler/src/rollup-plugins/module-resolver.ts (hunk 1 above)
Comment (ekashida) [reply packages/@lwc/compiler/src/rollup-plugins/module-resolver.ts]: We can either go with this or disable the eslint rule https://eslint.org/docs/rules/no-prototype-builtins

File: packages/integration-karma/test/polyfills/aria-properties/index.spec.js (hunk 2 above)
Comment (ekashida) [reply packages/integration-karma/test/polyfills/aria-properties/index.spec.js]: I didn't want to hook everything up to make @lwc/shared available in the karma env :)

**Pull request merged**